src/cli/README.md

```diff
@@ -3222,15 +3222,46 @@ offline, disabled, detached, child, router or leader
 Done
 ```
 
-### state <state>
+### state leader
+
+Become a leader and start a new partition
+
+If the device is not attached, this command will force the device to start as the leader of the network. This use case is only intended for testing and demo purposes, and using the API while the device is detached can make a production application non-compliant with the Thread Specification.
 
-Try to switch to state `detached`, `child`, `router` or `leader`.
+If the device is already attached, this API can be used to try to take over as the leader, creating a new partition. For this to work, the local leader weight (`leaderweight`) must be larger than the weight of the current leader (from `leaderdata`). If it is not, error `NotCapable` is outputted to indicate to the caller that they need to adjust the weight.
```

Comment:
To add similar disclaimer here that doing this without explicit user action can make device non-compliant.